app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -1191,6 +1234,209 @@ class BrowserTabFragment : Fragment(), FindListener, CoroutineScope {
+            } else {
+                decorator.decorateToolbarMenu(viewState)
+            }
+        }
+
+        private fun decorateToolbarMenu(viewState: BrowserViewState) {
+            tabsButton?.isVisible = viewState.showTabsButton
+            fireMenuButton?.isVisible = viewState.showFireButton
+            menuButton?.isVisible = viewState.showMenuButton
+        }
+
+        private fun hideToolbarMenu() {
+            menuButton?.gone()
+        }
+
+        fun decorateWithToolbarOnlyExperiment() {
+            hideBottomBar()
+            decorateAppBarWithToolbarOnlyExperiment()
+            createPopupMenuWithToolbarOnlyExperiment()
+            configureShowTabSwitcherListenerWithToolbarOnlyExperiment()
+            configureLongClickOpensNewTabListenerWithToolbarOnlyExperiment()
+        }
+
+        fun decorateWithBottomBarSearchExperiment() {
+            decorateBottomBarWithNavigationOnlyExperiment()
+            decorateAppBarWithBottomBarExperiment()
+            createPopupMenuWithBottomBarExperiment()
+            configureShowTabSwitcherListenerWithBottomBarNavigationOnlyExperiment()
+            configureLongClickOpensNewTabListenerWithBottomBarNavigationOnlyExperiment()
+        }
+
+        private fun decorateAppBarWithToolbarOnlyExperiment() {
+            toolbar.inflateMenu(R.menu.menu_browser_activity)
+            toolbar.setOnMenuItemClickListener { menuItem ->
+                when (menuItem.itemId) {
+                    R.id.fire -> {
+                        browserActivity?.launchFire()
+                        pixel.fire(String.format(Locale.US, Pixel.PixelName.MENU_ACTION_FIRE_PRESSED.pixelName, variantManager.getVariant()))
+                        return@setOnMenuItemClickListener true
+                    }
+                    else -> return@setOnMenuItemClickListener false
+                }
+            }
+        }
+
+        private fun decorateAppBarWithBottomBarExperiment() {
+            menuButton?.gone()
+        }
+
+        private fun createPopupMenuWithToolbarOnlyExperiment() {
+            popupMenu = BrowserPopupMenu(layoutInflater, variantManager.getVariant())
+            val view = popupMenu.contentView
+            popupMenu.apply {
+                onMenuItemClicked(view.forwardPopupMenuItem) { viewModel.onUserPressedForward() }
+                onMenuItemClicked(view.backPopupMenuItem) { activity?.onBackPressed() }
+                onMenuItemClicked(view.refreshPopupMenuItem) {
+                    viewModel.onRefreshRequested()
+                    pixel.fire(String.format(Locale.US, Pixel.PixelName.MENU_ACTION_REFRESH_PRESSED.pixelName, variantManager.getVariant()))
+                }
+                onMenuItemClicked(view.newTabPopupMenuItem) {
+                    viewModel.userRequestedOpeningNewTab()
+                    pixel.fire(String.format(Locale.US, Pixel.PixelName.MENU_ACTION_NEW_TAB_PRESSED.pixelName, variantManager.getVariant()))
```

Comment:
Should we move any `pixe.fire` inside the Viewmodel so we can test the logic?

Reply:
I thought about it, but then we need to add `VariantManager` as a collaborator just for the sake of testing it. Since this code will be removed once the experiment finishes I chose to go for the simple route and keep firing pixels at the `Activity` level. If we decide that Pixels must always be fired for these actions yes, definitely move it to the `ViewModel` and test it

Reply:
Fair point, I agree 👍  I liked that you considered both options. Thanks!